osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/main/java/org/opensmartgridplatform/adapter/protocol/dlms/application/services/GetKeysService.java

```diff
@@ -20,24 +20,26 @@
 import org.opensmartgridplatform.dto.valueobjects.smartmetering.KeyDto;
 import org.opensmartgridplatform.dto.valueobjects.smartmetering.SecretTypeDto;
 import org.opensmartgridplatform.shared.infra.jms.MessageMetadata;
-import org.opensmartgridplatform.shared.security.RsaEncryptionService;
+import org.opensmartgridplatform.shared.security.RsaEncrypter;
 import org.opensmartgridplatform.ws.schema.core.secret.management.SecretType;
 import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
 import org.springframework.stereotype.Component;
 
 @Component
 public class GetKeysService {
 
   private final SecretManagementService secretManagementService;
-  private final RsaEncryptionService keyEncrypter;
+  private final RsaEncrypter keyEncrypter;
```

Comment:
I saw issue [SMHE-15](https://alliander.atlassian.net/browse/SMHE-15) and I saw the comment at the top of the RsaEncrypter class, so I understand that RsaEncryptionService are going to be merged. However, I also read in SMHE-15 that there will be a single service with two instances with different configuration. Now I'm wondering: why is RsaEncryptionService changed into RsaEncrypter here? Is the configuration of RsaEncrypter better suited here?

Reply:
Yes, RsaEncrypter is better suited here. RsaEncryptionService has more functionality (e.g. checking if a private and public key match) which is not needed here. When we will work on issue SMHE-15 we can decide wheter the RsaEncryptionService can be completely removed, or that it has some functionality that has to be moved to the RsaEncrypter.
In the GetKeysService only 1 instance of the RsaEncrypter is needed. In other places two instances are needed, one for decrypting with the private key and one for encrypting with the public key of another component. (That's also why the check on matching keys is not necessary: a component has it's own private key to decypt incoming messages and a public key to encrypt messages to another component, so these keys do no match).